lib/rouge/formatters/terminal256.rb

```diff
@@ -85,6 +82,14 @@ def bg
           @bg = style.bg && self.class.color_index(style.bg)
         end
 
+
+        def stream_value(val, &b)
+          yield style_string
+          yield val.gsub("\n", "#{reset_string}\n#{style_string}")
+                   .gsub("\e", "\\e")
```

Comment:
oops, these gsubs should definitely be in reverse order, since `reset_string` and `style_string` will very likely contain `\e`

Reply:
Note for posterity that this was fixed in #1404.